src/server/ua_services_discovery.c

```diff
@@ -118,3 +232,65 @@ void Service_GetEndpoints(UA_Server *server, UA_Session *session, const UA_GetEn
         return;
     }
 }
+
+
+void Service_RegisterServer(UA_Server *server, UA_Session *session,
+                         const UA_RegisterServerRequest *request, UA_RegisterServerResponse *response) {
+    UA_LOG_DEBUG_SESSION(server->config.logger, session, "Processing RegisterServerRequest");
+
+    registeredServer_list_entry *registeredServer_entry = NULL;
+
+    {
```

Comment:
I added the extra block to hide the `current` variable created in the next line from the rest of the code to keep it clean.

Is that not needed?